src/app/api/auth/[auth0]/route.ts

```diff
@@ -0,0 +1,4 @@
+/* istanbul ignore file */
+import { auth0 } from "@/node-lib/auth/auth0";
+
+export const GET = auth0.handleAuth();
```

Comment:
This installs a bunch of API routes.

* `/api/auth/login` — which we can link to in the app to send the user on their login/signup journey
* `/api/auth/callback` — this is the page the user is returned to after their journey, it is the only endpoint authorised in Auth0 to receive id and access tokens
* `/api/auth/logout` — hitting this will log the user out in the app (but not destroy their session in Auth0!)
* `/api/auth/me` — this is the endpoint `UserProvider` will hit to get the logged in user's profile